pandas/tests/resample/test_datetime_index.py

```diff
@@ -1488,3 +1488,13 @@ def test_get_timestamp_range_edges(first, last, offset,
     result = _get_timestamp_range_edges(first, last, offset)
     expected = (exp_first, exp_last)
     assert result == expected
+
+
+def test_apply_on_series():
+    # GH 25959
+    # Calling apply on a localized time series should not cause an error
```

Comment:
pls parameterize over both the working and the originally non-working example in the OP

Reply:
Fixed!